add editable custom model

B1ueRaw opened the pull request:
Now users can free to edit their custom models by double clicking any custom models created in each provider.
Before this, they have to delete -> create a new one.
I simply add an edit panel (which looks the same as 'Add Custom Model') and integrate that process into the "update" button.

There is one more issue that if a user deletes a model that he was using in chat, then back to chat, that model would still appear (and work) unless user chooses a new one.
Tried to modify "delete-custom-model" in language_model_handlers.ts by the logic that if the name of that model matches the latest using one -> switch to auto (or default) model. Yet I failed, maybe need more explanation for this :)

B1ueRaw committed:
add editable custom model panel

B1ueRaw committed:
update 'update model' logic

Changed the update logic in EditCustomModelDialog to delete the old custom model and create a new one instead of updating in place. Removed unnecessary resetForm calls to streamline dialog closing.

B1ueRaw committed:
Remove unused function

wwwillchen commented:
Thanks for opening this PR. Can you add an E2E test to cover this? See https://github.com/dyad-sh/dyad/blob/main/CONTRIBUTING.md#e2e-tests

We already have some tests for this feature area, e.g. https://github.com/dyad-sh/dyad/blob/main/e2e-tests/delete_provider.spec.ts

B1ueRaw committed:
add E2E test cover edit custom panel

two cases:
1. double clicking the custom model panel to edit
2. click the edit button to edit

B1ueRaw commented:
> Thanks for opening this PR. Can you add an E2E test to cover this? See https://github.com/dyad-sh/dyad/blob/main/CONTRIBUTING.md#e2e-tests
> 
> We already have some tests for this feature area, e.g. https://github.com/dyad-sh/dyad/blob/main/e2e-tests/delete_provider.spec.ts

Two cases were added since I expect users to jump into the edit panel by two ways:
1. Double click the whole custom block area
2. click the edit button

wwwillchen commented:
LGTM - thanks!
